src/NetPanzer/Interfaces/MapInterface.cpp

```diff
@@ -55,11 +55,11 @@ bool MapInterface::startMapLoad( const char *file_path, const char *mapstyle_mpa
     NPString stpath = mapstyle_mpath;
     NPString tsname = main_map.getAssocTileSet();
 
-    char subbuff[24];
-    memcpy( subbuff, &tsname[0], 23); // max 23 chars for tileset name
-    char * pch;
-    pch = strstr (subbuff,".tls");
-    strncpy (pch,"\0",1);
+    const int max_len_tsname = 23;
+    std::string subbuff = tsname.substr(0, max_len_tsname);
+    size_t found = subbuff.find(".tls");
+    if (found <= max_len_tsname)
```

Comment:
hm, find returns -1 when no match, but i guess since it's size_t it "overflows" to largest value of size_t and so this condition is false?

Reply:
I'm not completely confident there.

https://cplusplus.com/reference/string/string/find/

>If no matches were found, the function returns [string::npos](https://cplusplus.com/string::npos).

If I interpret that right, it would return a very large value if the substring isn't found.

Reply:
in practice - yes - not saying there is an issue. just learning :)

Reply:
Me too. I'm glad you questioned it. Feel free to do so any time you're not sure.

This is a bit ridiculous though: I tested and printed the return value of 'found' to stderr. Showed '-1'. 

So I changed the code and got this warning

```
../src/NetPanzer/Interfaces/MapInterface.cpp:61:15: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
   61 |     if (found != -1)
```
So even though I declared 'found' as a size_t, it returned a signed int, which should be unsigned (where ssize_t can be signed). As far as I know. ;)